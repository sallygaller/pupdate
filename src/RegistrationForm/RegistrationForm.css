.RegistrationForm {
    display: flex;
    flex-direction: column;
    text-align: left;
    margin: 0 auto;
    width: 30%;
}

.RegistrationForm input {
    width: 100%;
    padding: 12px 20px;
    margin: 8px 0;
    display: inline-block;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-sizing: border-box;
  }

  .RegistrationForm button {
    width: 100%;
    background-color: #4CAF50;
    color: white;
    padding: 14px 20px;
    margin: 8px 0;
    border: none;
    border-radius: 4px;
    cursor: pointer;
}

.RegistrationForm button:enabled:hover {
    background-color: #45a049;
}

.RegistrationForm-error {
    color: red;
}