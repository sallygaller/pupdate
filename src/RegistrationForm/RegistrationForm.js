import React, { useState } from "react";
import SearchCityState from "../SearchCityState/SearchCityState";
import AuthApiService from "../services/auth-api-service";
import "./RegistrationForm.css";

export default function RegistrationForm(props) {
  const [firstname, setFirstname] = useState("");
  const [lastname, setLastname] = useState("");
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [location, setLocation] = useState("");
  const [error, setError] = useState(null);

  const Required = () => <span>*</span>;

  const handleSubmit = (e) => {
    e.preventDefault();
    if (location === "") {
      setError("Please complete the 'City, State, Country' field below.");
    } else {
      let locationArray = location.split(",");
      const user = {
        firstname,
        lastname,
        email,
        password,
        city: locationArray[0],
        state: locationArray[1].trim(),
      };
      AuthApiService.postUser(user)
        .then((user) => {
          props.onRegistrationSuccess();
        })
        .catch((res) => {
          setError(res.error);
        });
    }
  };

  return (
    <form className="RegistrationForm" onSubmit={(e) => handleSubmit(e)}>
      <div className="RegistrationForm-error" role="alert">
        {error && (
          <p>
            {error.message}
            {error}
          </p>
        )}
      </div>
      <label htmlFor="first-name">
        First Name: <Required />
      </label>
      <input
        required
        type="text"
        value={firstname}
        id="first-name"
        onChange={(e) => setFirstname(e.target.value)}
      />
      <label htmlFor="last-name">
        Last Name: <Required />{" "}
      </label>
      <input
        required
        type="text"
        name="last-name"
        value={lastname}
        id="last-name"
        onChange={(e) => setLastname(e.target.value)}
      />
      <label htmlFor="email">
        Email: <Required />
      </label>
      <input
        required
        type="text"
        name="email"
        id="email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <label htmlFor="password">
        Password: <Required />
      </label>
      <p>
        (must be longer than 8 characters and contain 1 upper case, lower case,
        number and special character)
      </p>
      <input
        required
        type="password"
        name="password"
        id="password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <label htmlFor="zip">
        City, State, Country: <Required />
      </label>{" "}
      <SearchCityState
        location={location}
        setLocation={setLocation}
        placeholder={"Portland, OR, USA"}
      />
      <button type="submit">Sign Up!</button>
    </form>
  );
}
