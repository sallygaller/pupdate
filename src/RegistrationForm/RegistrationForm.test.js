import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import RegistrationForm from "./RegistrationForm";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <RegistrationForm />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer.create(<RegistrationForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
