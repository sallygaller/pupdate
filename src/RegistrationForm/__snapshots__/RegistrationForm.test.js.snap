// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div
    className="RegistrationForm-error"
    role="alert"
  />
  <label
    htmlFor="first-name"
  >
    First Name: 
  </label>
  <input
    id="first-name"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="last-name"
  >
    Last Name: 
  </label>
  <input
    id="last-name"
    name="last-name"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="email"
  >
    Email: 
  </label>
  <input
    id="email"
    name="email"
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="password"
  >
    Password: 
  </label>
  <input
    id="password"
    name="password"
    onChange={[Function]}
    required={true}
    type="password"
    value=""
  />
  <label
    htmlFor="zip"
  >
    City and State: 
  </label>
   
  <div
    className="search-location-input"
  >
    <input
      onChange={[Function]}
      placeholder="Begin typing a city and state"
      value=""
    />
  </div>
  <button
    type="submit"
  >
    Sign Up!
  </button>
</form>
`;
